src/lib/storage/table.cpp

```diff
@@ -1,84 +1,108 @@
+      _chunks.back()->add_segment(value_segment);
+    });
+  }
 }
 
 void Table::append(const std::vector<AllTypeVariant>& values) {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  if (_chunks.back()->size() == _chunk_size) {
+    create_new_chunk();
+  }
+  _chunks.back()->append(values);
 }
 
 ColumnCount Table::column_count() const {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  return static_cast<ColumnCount>(_column_names.size());
 }
 
 uint64_t Table::row_count() const {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  return (_chunks.size() - 1) * _chunk_size + _chunks.back()->size();
 }
 
 ChunkID Table::chunk_count() const {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  return static_cast<ChunkID>(_chunks.size());
 }
 
 ColumnID Table::column_id_by_name(const std::string& column_name) const {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  for (unsigned int i = 0; i < _column_names.size(); i++) {
+    if (_column_names[i] == column_name) {
+      return static_cast<ColumnID>(i);
+    }
+  }
+  throw std::logic_error("Table does not contain column with the requested name.");
 }
 
 ChunkOffset Table::target_chunk_size() const {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  return _chunk_size;
 }
 
 const std::vector<std::string>& Table::column_names() const {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  return _column_names;
 }
 
 const std::string& Table::column_name(const ColumnID column_id) const {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  if (column_id > _column_nullable.size()) {
+    throw std::logic_error("Table does not contain column with the requested id.");
+  }
+  return _column_names[column_id];
 }
 
 const std::string& Table::column_type(const ColumnID column_id) const {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  if (column_id > _column_nullable.size()) {
+    throw std::logic_error("Table does not contain column with the requested id.");
+  }
+  return _column_types[column_id];
 }
 
 bool Table::column_nullable(const ColumnID column_id) const {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  if (column_id > _column_nullable.size()) {
+    throw std::logic_error("Table does not contain column with the requested id.");
+  }
+  return _column_nullable[column_id];
 }
 
 std::shared_ptr<Chunk> Table::get_chunk(ChunkID chunk_id) {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  if (chunk_id >= static_cast<ChunkID>(_chunks.size())) {
+    throw std::logic_error("Table does not contain chunk with the requested id.");
+  }
+  return _chunks[chunk_id];
 }
 
 std::shared_ptr<const Chunk> Table::get_chunk(ChunkID chunk_id) const {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  return _chunks[chunk_id];
 }
```

Comment:
Is the upper method supposed to return a copy, perhaps?
why else would we distinguish between const and not const?

Reply:
Chunk is a subclass of Noncopyable... Actually, I don't know what the difference between those to methods should be...